lib/util/Components.js

```diff
@@ -855,11 +855,6 @@ function componentRule(rule, context) {
     },
 
     FunctionExpression(node) {
-      if (node.async) {
-        components.add(node, 0);
-        return;
```

Comment:
instead of just allowing any async functions - are there any static aspects of async components that can help ensure we're not going to just grab a bunch of non-component async functions?

for example, i assume an async generator isn't allowed to be a component, so we should still check for that?